### ISSUE TITLE
SentryHttpFailedRequestHandler disposes Content on .net framework

### ISSUE THREAD

**Issue by Liklainy** [getsentry/sentry-dotnet#2684]
### Package

Sentry

### .NET Flavor

.NET Framework

### .NET Version

4.7.2

### OS

Any (not platform specific)

### SDK Version

3.33.0

### Steps to Reproduce

1. Generate client from OpenApi with `NSwag`
2. Create `HttpClient` with `SentryHttpMessageHandler` and pass it to generated client
3. Send request which will fail and will be caught with `SentryHttpFailedRequestHandler`

### Expected Result

No exception thrown

### Actual Result

1. Handler uses `EnsureSuccessStatusCode` which disposes `Content` on older versions (before net core)
2. NSwag generated code checks status code and if it's not `200` it tries to read content body which is already disposed and generates `ObjectDisposed` exception

**Comment by jamescrosswell** [comment]
Hi @Liklainy, thanks for getting in touch. 

@bitsandfoxes we could fairly easily implement our own `EnsureSuccessStatusCode` for target frameworks below `.net5.0`. I know Matt was against this on principal, as it would mean we wouldn't get any fixes if the code we copied changed. 

However, we've already got a [workaround for the broken framework code](https://github.com/getsentry/sentry-dotnet/blob/b6ac6e9500222d1e66b3de595b5ca712702851b3/src/Sentry/SentryHttpFailedRequestHandler.cs#L23-L34), but at the moment that workaround only protects the proper functioning of our own code (but doesn't guarantee other stuff won't get broken).

In this particular case, I'd be OK to bend the "don't copy framework code into our solution" rule, as long as our code is only used on .netcore3.1 and below. Under the hood, all `EnsureSuccessStatusCode` does is throw an exception if the response status code isn't in the 200 range, so trivial to implement ourselves:
* https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System/net/System/Net/Http/HttpResponseMessage.cs#L117-L120
* https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/System/net/System/Net/Http/HttpResponseMessage.cs#L142-L159
